service/worker/workerdeployment/version_workflow.go

```diff
@@ -243,6 +245,12 @@ func (d *VersionWorkflowRunner) startDrainage(ctx workflow.Context, isCan bool)
 	d.drainageWorkflowFuture = &fut
```

Comment:
Not this PR but I just happened to see it: the future is an interface type, you should just use it directly, not use a pointer to it